Source/ASTextNode.mm

```diff
@@ -85,8 +88,13 @@ - (BOOL)isEqual:(ASTextNodeRendererKey *)object
   if (self == object) {
     return YES;
   }
+
+  // Lock both objects, avoiding deadlock.
+  std::lock(_m, object->_m);
```

Comment:
Unrelated to you changes, should there be class checks in here? It may not matter in since this is an internal object, but should we ensure object is an ASTextRendererKey for correctness?

Reply:
Good call. I added a comment explaining it.